pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -607,13 +607,57 @@ func validateMachinePools(cd *hivev1.ClusterDeployment) field.ErrorList {
 
 	vErrs = append(vErrs, validateMachinePool(cd, &cd.Spec.ControlPlane, field.NewPath("spec", "controlPlane"))...)
 
+	vErrs = append(vErrs, validateComputeMachinePoolRequirements(cd)...)
+
 	for i, mp := range cd.Spec.Compute {
 		vErrs = append(vErrs, validateMachinePool(cd, &mp, field.NewPath("spec", "compute").Index(i))...)
 	}
 
 	return vErrs
 }
 
+func validateComputeMachinePoolRequirements(cd *hivev1.ClusterDeployment) field.ErrorList {
+	vErrs := field.ErrorList{}
+	path := field.NewPath("spec")
+
+	if cd.Spec.Platform.GCP != nil {
+		// Only restrict the length of compute machine pools on GCP.
+		// Restriction can be removed once proper handling of reuse or creation
+		// of necessary subnets is resolved.
+		if len(cd.Spec.Compute) > 1 {
```

Comment:
Collapse both of these if blocks into a simpler validation that the name is "worker".

```
for i, mp := range cd.Spec.Compute {
  if mp.Name != "worker" {
    vErrs = append(vErrs, field.Invalid(computePath.Index(i).Child("name"), mp.Name, "name of the compute pool must be \"worker\"")
  }
}
```

If we only allow "worker" pools and the names must be unique, then there cannot be more than one pool.

If there can only be one pool and the only valid name is "worker", then if we have more than zero pools, then there must be a pool named "worker".